### PR TITLE
Fix sync behaviour for imported files

### PR DESCRIPTION
**Pull request by stalleyj**
## What type of PR is this ? 

- [X] Bug fix
- [ ] Enhancement

## What does this PR do ?
This PR fixes our syncing behaviour to identify files that have been imported into the project via the Eclipse IDE.  When a user does this, the modified time of the file does not get updated so our current sync checks for last mod time > last sync time don't pick up the new files.  

The new behaviour looks to see if each file contained in the project (on the users disk) is in the list of files for the project (that have been uploaded to codewind). If that file doesn't exist, it will be uploaded. 

## Which issue(s) does this PR fix ?

https://github.com/eclipse/codewind/issues/2860

## Does this PR require a documentation change ?
No

## Any special notes for your reviewer ?
No

### CODE REVIEW

File: pkg/project/filelist.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	if readErr != nil {
+		return nil, &ProjectError{errOpRequest, readErr, readErr.Error()}
+	}
+
+	var list FileList
+	jsonErr := json.Unmarshal(byteArray, &list)
+	if jsonErr != nil {
+		return nil, &ProjectError{errOpRequest, jsonErr, jsonErr.Error()}
+	}
+	//	fmt.Println(list)
```
Comment (hhellyer) [inline pkg/project/filelist.go:45]: Does this need tidying? (Commented out `Println`.)

Comment (stalleyj) [reply pkg/project/filelist.go]: Done

File: pkg/project/sync.go [hunk 2]
```diff
@@ -457,3 +429,79 @@ func handleMissingProjectDir(httpClient utils.HTTPClient, connection *connection
...
+	zWriter.Close()
+	encoded := base64.StdEncoding.EncodeToString(buffer.Bytes())
+	fileUploadBody.Message = encoded
+
+	buf := new(bytes.Buffer)
+	json.NewEncoder(buf).Encode(fileUploadBody)
+
+	projectUploadURL := conURL + "/api/v1/projects/" + projectID + "/upload"
+	// TODO - How do we handle partial success?
+	request, err := http.NewRequest("PUT", projectUploadURL, bytes.NewReader(buf.Bytes()))
```
Comment (hhellyer) [inline pkg/project/sync.go:494]: I don't think it's likely to happen and this isn't new code but we don't check `err` here.

**Comment by mattcolegate** [comment]
Tested on Windows using recreation instructions from https://github.com/eclipse/codewind/issues/2860 - imported hello/Hello.java and project rebuilt and I was able to run the new application from the /hello endpoint

File: pkg/project/sync.go (hunk 2 above)
Comment (stalleyj) [reply pkg/project/sync.go:494]: I tried not to change anything if I didn't need to, can be addressed later

**Pull request merged**